android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -392,21 +391,41 @@ constructor(
     defaultRepository.addOrUpdate(resource = questionnaireResponse)
   }
 
-  suspend fun performExtraction(
+  fun performExtraction(
     context: Context,
     questionnaire: Questionnaire,
     questionnaireResponse: QuestionnaireResponse
   ): Bundle {
-
-    return ResourceMapper.extract(
-      questionnaire = questionnaire,
-      questionnaireResponse = questionnaireResponse,
-      StructureMapExtractionContext(
-        context = context,
-        transformSupportServices = transformSupportServices,
-        structureMapProvider = retrieveStructureMapProvider()
-      )
-    )
+    var bundle = Bundle()
+    viewModelScope.launch(dispatcherProvider.main()) {
+      kotlin
+        .runCatching {
+          withContext(dispatcherProvider.io()) {
+            ResourceMapper.extract(
+              questionnaire = questionnaire,
+              questionnaireResponse = questionnaireResponse,
+              StructureMapExtractionContext(
+                context = context,
+                transformSupportServices = transformSupportServices,
+                structureMapProvider = retrieveStructureMapProvider()
+              )
+            )
+          }
+        }
+        .onSuccess {
+          bundle = it
+          context.showToast(context.getString(R.string.structure_success))
+        }
+        .onFailure { exception ->
+          Timber.e(exception)
+          if (exception.message!!.contains("StructureMap")) {
+            context.showToast(context.getString(R.string.structure_map_missing_message))
```

Comment:
can we mention the specific structuremap it was looking for in the error?

Reply:
We want some info so that we can track down what's the causing the problem, maybe the form name would be enough, if we can only have a single structuremap tied to a form, but if we can have more then 1 structuremap tied to a form, we'd want a way to tell which

Reply:
Added questionnaire name to as part of the toast message. This is because in case of exception, the function  `ResourceMapper.extract` returns an empty bundle and hence we cannot know the StructureMap name.